api/clever/__init__.py

```diff
@@ -154,6 +162,13 @@ def oauth_callback(self, _db, params):
         )
         patron._external_type = external_type
 
+        credential, is_new = get_one_or_create(
+            _db, Credential, data_source=self._data_source(_db),
+            type=self.TOKEN_TYPE, patron=patron,
+        )
+        credential.credential = token
+        credential.expires = datetime.datetime.utcnow() + datetime.timedelta(weeks=6)
```

Comment:
6 weeks is a good default but it should be configurable. (Even though it's easier for us to deploy a code change than a configuration change.)